functions-and-scope.js

```diff
@@ -69,15 +133,35 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // * Op welke conditie moet ik checken?
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
 // Log het antwoord in de terminal.
+console.log ("------------------ Opdracht 3a -------------------")
+console.log ("Iedere waarden kan je langsgaan door het gebruiken van van een for loop");
+console.log ("Of het cijfer hoger is dan de vorige");
+console.log ("Als het cijfer hoger is mag deze de variable overschijven")
 
 // ---- Verwachte uitkomst: 9
 
+console.log ("------------------ Opdracht 3b -------------------")
 
 /* 3b: Omschrijven tot een herbruikbare functie */
 // Schrijf een functie genaamd highestGrade, die een array van cijfers verwacht (zoals grades) en het hoogste cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 3a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
+function highestGrade (input){
+    let highest = 0;
```

Comment:
Wanneer je met een for-loop gaat bepalen wat de hoogste (of laagste) waarde is, dan is het beter om hier de eerste waarde uit de array te pakken
```suggestion
    let highest = input[0];
```

Met rapportcijfers maakt het niet zo veel uit, maar stel je voor dat je deze functie had aangeroepen met de array `[-1, -3, -8, -4]`. Dan had de uitkomst van deze functie `0` geweest terwijl die waarde niet eens in de array voorkomt.